tests/test_greedy_selector.py

```diff
@@ -0,0 +1,96 @@
+import unittest
+import numpy as np
+
+from sklearn.datasets import load_boston
+from sklearn import exceptions
+from sklearn.exceptions import NotFittedError
+
+from skcosmo.feature_selection._greedy import GreedySelector
+
+
+class GreedyTester(GreedySelector):
+    def __init__(self, n_features_to_select=None, score_thresh_to_select=None):
+        super().__init__(
+            scoring=self.score,
+            n_features_to_select=n_features_to_select,
+            score_thresh_to_select=score_thresh_to_select,
+        )
+
+    def score(self, X, y=None):
+        scores = np.linalg.norm(X, axis=0)
+        scores[self.selected_idx_] = 0.0
+        return scores
+
+
+class TestGreedy(unittest.TestCase):
+    def setUp(self):
+        self.X, _ = load_boston(return_X_y=True)
+
+    def test_score_threshold(self):
+        selector = GreedyTester(score_thresh_to_select=20, n_features_to_select=12)
+        with self.assertWarns(
+            Warning, msg="Score threshold of 20 reached. Terminating search at 10 / 12."
```

Comment:
Why does it stop at 10/12 features if the maximum value is 20?

Reply:
the minimum value for the score is 20, i.e. "Choose as many features as have a score greater than 20", which here happens to be 10.

Reply:
Oh, yep, I were slightly confused, thank you!